### ISSUE TITLE
Alias ignored when flattening

### ISSUE THREAD

**Issue by TheEnbyperor** [serde-rs/serde#2188]
Given an enum with a name and an alias, such as below.
```rust
#[derive(Debug, Serialize, Deserialize, Clone)]
pub enum PSCNatureOfControls {
    #[serde(
        rename = "{http://xmlgw.companieshouse.gov.uk}NatureOfControls",
        alias = "{http://www.govtalk.gov.uk/CM/envelope}NatureOfControls"
    )]
    NatureOfControls(NatureOfControls),
    #[serde(
        rename = "{http://xmlgw.companieshouse.gov.uk}LLPNatureOfControls",
        alias = "{http://www.govtalk.gov.uk/CM/envelope}LLPNatureOfControls"
    )]
    LLPNatureOfControls(LLPNatureOfControls),
}
```

Then put in a struct as such, and flattened.
```rust
#[derive(Debug, Deserialize)]
pub struct CompanyDataPSCNotification {
    #[serde(flatten)]
    pub nature_of_control: PSCNatureOfControls,
    ...
}
```

And when an enum value with the alias name is presented to the deserializer, an error such as "no variant of enum PSCNatureOfControls found in flattened data" is given. When the enum variant is changed from alias to rename, it works as expected.

**Comment by jplatte** [comment]
Duplicate of #1504.

**Comment by Mingun** [comment]
This was fixed (partially) by #2387, the in-place part is fixed in #2443
